src/contrib/inference/dynamichmc.jl

```diff
@@ -60,8 +60,8 @@ function DynamicPPL.initialstep(
 )
     # Ensure that initial sample is in unconstrained space.
     if !DynamicPPL.islinked(vi, spl)
-        DynamicPPL.link!(vi, spl)
-        model(rng, vi, spl)
+        vi = DynamicPPL.link!!(vi, spl, model)
+        vi = last(DynamicPPL.evaluate!!(model, vi, DynamicPPL.SamplingContext(rng, spl)))
```

Comment:
Can you remind me why we run the model here?

Reply:
And why we use the `SamplingContext`?

Reply:
I _think_ this was done to update the logp after `link!`. _But_ now `link!!` also includes the log-absdet-jacobian correction, so we could potentially drop it.

And regarding why we use the `SamplingContext`: didn't want to make changes to something I wasn't 100% certain would have no side-effects. But yes, I also think this is no longer necessary.

The question is if we should make these changes _now_ or in another PR.

Reply:
Maybe safer to move these questions in a separate issue and potentially address them in a separate PR :+1: